runelite-client/src/main/java/net/runelite/client/plugins/worldmap/mappoints/FairyRingPoint.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2018, Morgan Lewis <https://github.com/MESLewis>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.worldmap.mappoints;
+
+import com.google.common.collect.ImmutableSet;
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.util.Set;
+import javax.imageio.ImageIO;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.plugins.worldmap.WorldMapPlugin;
+import net.runelite.client.ui.overlay.worldmap.WorldMapPoint;
+
+public class FairyRingPoint extends IconWorldMapPoint
+{
+	private final static BufferedImage FAIRY_RING_ICON;
+
+	static
+	{
+			//A size of 17 gives us a buffer when triggering tooltips
+			final int ICON_BUFFER_SIZE = 17;
+			FAIRY_RING_ICON = new BufferedImage(ICON_BUFFER_SIZE, ICON_BUFFER_SIZE, BufferedImage.TYPE_INT_ARGB);
+	}
+
+	public static void setCustomIcon(boolean customIcon)
+	{
+		if (customIcon)
+		{
+			try
+			{
+				synchronized (ImageIO.class)
+				{
+					final BufferedImage icon = ImageIO.read(WorldMapPlugin.class.getResourceAsStream("fairy_ring_travel.png"));
+					FAIRY_RING_ICON.getGraphics().drawImage(icon, 1, 1, null);
+				}
+			}
+			catch (IOException e)
+			{
+				throw new RuntimeException(e);
+			}
+		}
+		else
+		{
+			Graphics2D iconGraphics = (Graphics2D) FAIRY_RING_ICON.getGraphics();
+			iconGraphics.setBackground(new Color(0xFFFFFF, true));
+			iconGraphics.clearRect(0, 0, FAIRY_RING_ICON.getWidth(), FAIRY_RING_ICON.getHeight());
+		}
+	}
+
+	private static final Set<WorldMapPoint> POINTS = ImmutableSet.of(
```

Comment:
THis should probably be enum

Reply:
I'm following what the clue scroll plugin does for it's sets of clues.

Reply:
You should probably rather loop through all the points and set the image, this is really really ugly way to do this.

Reply:
Maybe creating the FairyRingPoint from some enum that contains just the string and worldpoint when needed will solve this, then you can use config to use correct icon, and you can store this as data properly.

Reply:
I could loop through each FairyRingPoint but they will always all display the same thing so I just pass in the reference to the single image. 
I could have an icon image and a blank image (needed for the hitbox) and loop through each FairyRingPoint when the setting is changed and set it to the new image but that seems pointless when they all share an image anyway.

Reply:
Yeah, that is a lot better than current implementation of this.

Reply:
I don't see how it would be better though. There is no reason to loop through every point when you can just draw/clear the shared image directly.

Reply:
Drawing to shared image is bad thing to do is what I am saying. That is just abusing the BufferedImage. It is mostly about having non-bad implementation of this, and looping through is more natural thing to do. Even though you are represeting your data as constants, but still modifying it, so even setting the image would be wrong. I would go with my other suggestion and that is creating the map points from some enum.